src/brsTypes/components/RoDeviceInfo.ts

```diff
@@ -0,0 +1,687 @@
+import { BrsComponent } from "./BrsComponent";
+import { Callable, StdlibArgument } from "../Callable";
+import { Interpreter } from "../../interpreter";
+import { RoAssociativeArray, AAMember } from "./RoAssociativeArray";
+
+export class RoDeviceInfo extends BrsComponent implements BrsValue {
+    readonly kind = ValueKind.Object;
+
+    constructor() {
+        super("roDeviceInfo");
+        this.registerMethods({
+            ifDeviceInfo: [
+                this.getModel,
+                this.getModelDisplayName,
+                this.getModelType,
+                this.getModelDetails,
+                this.getFriendlyName,
+                this.getOSVersion,
+                this.getVersion,
+                this.getRIDA,
+                this.isRIDADisabled,
+                this.getChannelClientId,
+                this.getUserCountryCode,
+                this.getRandomUUID,
+                this.getTimeZone,
+                this.hasFeature,
+                this.getCurrentLocale,
+                this.getCountryCode,
+                this.getPreferredCaptionLanguage,
+                this.timeSinceLastKeyPress,
+                this.getDrmInfo,
+                this.getDrmInfoEx,
+                this.getCaptionsMode,
+                this.setCaptionsMode,
+                this.getCaptionsOption,
+                this.getClockFormat,
+                this.enableAppFocusEvent,
+                this.enableScreensaverExitedEvent,
+                this.enableLowGeneralMemoryEvent,
+                this.getGeneralMemoryLevel,
+                this.isStoreDemoMode,
+                this.getLinkStatus,
+                this.enableLinkStatusEvent,
+                this.getConnectionType,
+                this.getExternalIp,
+                this.getIPAddrs,
+                this.getConnectionInfo,
+                this.getDisplayType,
+                this.getDisplayMode,
+                this.getDisplayAspectRatio,
+                this.getDisplaySize,
+                this.getVideoMode,
+                this.getDisplayProperties,
+                this.getSupportedGraphicsResolutions,
+                this.canDecodeVideo,
+                this.getUIResolution,
+                this.getGraphicsPlatform,
+                this.enableCodecCapChangedEvent,
+                this.getAudioOutputChannel,
+                this.getAudioDecoderInfo,
+                this.canDecodeAudio,
+                this.getSoundEffectsVolume,
+                this.isAudioGuideEnabled,
+                this.enableAudioGuideChangedEvent,
+            ],
+        });
+    }
+
+    toString(parent?: BrsType): string {
+        return "<Component: roDeviceInfo>";
+    }
+
+    equalTo(other: BrsType) {
+        return BrsBoolean.False;
+    }
+
+    private getModel = new Callable("getModel", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("4280x");
```

Comment:
Hmmm i'm a little hesitant to return this type of static data for all of these functions. I'd definitely prefer to return empty strings (`BrsString("")`), because otherwise it would require users to _know_ these "seeded" values so they could expect them in unit tests.

Curious to hear what @sjbarag thinks as well.